pkg/metallb/frr_test.go

```diff
@@ -0,0 +1,440 @@
+		assert.Equal(t, testCase.expectedError, frrConfigurationBuilder.errorMsg)
+
+		if testCase.expectedError == "" {
+			assert.Equal(t, testCase.bgpPassword,
+				frrConfigurationBuilder.Definition.Spec.BGP.Routers[0].Neighbors[0].Password)
+		}
+	}
+}
+
+func TestFrrConfigurationWithHoldTime(t *testing.T) {
+	testCases := []struct {
+		testFrrConfiguration *FrrConfigurationBuilder
+		holdTime             *metav1.Duration
+		expectedError        string
+	}{
+		{
+			testFrrConfiguration: buildValidFRRConfigurationBuilder(buildFrrConfigurationTestClientWithDummyObject()),
+			holdTime: &metav1.Duration{
+				Duration: 90 * time.Second,
+			},
+		},
+		{
+			testFrrConfiguration: buildInValidFrrConfigurationBuilder(buildFrrConfigurationTestClientWithDummyObject()),
+			holdTime: &metav1.Duration{
+				Duration: 0 * time.Minute,
+			},
+			expectedError: "frrConfiguration 'peerIP' of the BGPPeer contains invalid ip address",
+		},
+	}
+
+	for _, testCase := range testCases {
+		frrConfigurationBuilder := testCase.testFrrConfiguration.WithHoldTime(*testCase.holdTime)
+		assert.Equal(t, testCase.expectedError, frrConfigurationBuilder.errorMsg)
+
+		if testCase.expectedError == "" {
+			assert.Equal(t, testCase.holdTime,
+				frrConfigurationBuilder.Definition.Spec.BGP.Routers[0].Neighbors[0].HoldTime)
+		}
+	}
+}
+
+func TestFrrConfigurationWithKeepalive(t *testing.T) {
+	testCases := []struct {
+		testFrrConfiguration *FrrConfigurationBuilder
+		keepalive            *metav1.Duration
+		expectedError        string
+	}{
+		{
+			testFrrConfiguration: buildValidFRRConfigurationBuilder(buildFrrConfigurationTestClientWithDummyObject()),
+			keepalive: &metav1.Duration{
+				Duration: 30 * time.Second,
+			},
+		},
+		{
+			testFrrConfiguration: buildInValidFrrConfigurationBuilder(buildFrrConfigurationTestClientWithDummyObject()),
+			keepalive: &metav1.Duration{
+				Duration: 0 * time.Second,
+			},
+			expectedError: "frrConfiguration 'peerIP' of the BGPPeer contains invalid ip address",
+		},
+	}
+
+	for _, testCase := range testCases {
+		frrConfigurationBuilder := testCase.testFrrConfiguration.WithKeepalive(*testCase.keepalive)
+		assert.Equal(t, testCase.expectedError, frrConfigurationBuilder.errorMsg)
+
+		if testCase.expectedError == "" {
+			assert.Equal(t, testCase.keepalive,
+				frrConfigurationBuilder.Definition.Spec.BGP.Routers[0].Neighbors[0].KeepaliveTime)
+		}
+	}
+}
+
+func TestFrrConfigurationWithConnectTime(t *testing.T) {
+	testCases := []struct {
+		testFrrConfiguration *FrrConfigurationBuilder
+		connectTime          *metav1.Duration
+		expectedError        string
+	}{
+		{
+			testFrrConfiguration: buildValidFRRConfigurationBuilder(buildFrrConfigurationTestClientWithDummyObject()),
+			connectTime: &metav1.Duration{
+				Duration: 120 * time.Second,
+			},
+		},
+		{
+			testFrrConfiguration: buildInValidFrrConfigurationBuilder(buildFrrConfigurationTestClientWithDummyObject()),
```

Comment:
please add tests for the valid but with Connect time 0 and more than  65535

Reply:
Added.